tickit/devices/eiger/eiger.py

```diff
@@ -0,0 +1,453 @@
+            "header_detail": header_detail,
+        }
+        if header_detail != "none":
+            config_json = {}
+            disallowed_configs = ["flatfield", "pixelmask" "countrate_correction_table"]
+            for field_ in fields(self.settings):
+                if field_.name not in disallowed_configs:
+                    config_json[field_.name] = vars(self.settings)[field_.name]
+
+        LOGGER.debug(json)
+        LOGGER.debug(config_json)
+
+    async def disarm(self) -> None:
+        """Function to disarm the Eiger."""
+        self._set_state(State.IDLE)
+
+        json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(json)
+
+    async def trigger(self) -> str:
+        """Function to trigger the Eiger.
+
+        If the detector is in an external trigger mode, this is disabled as
+        this software command interface only works for internal triggers.
+        """
+        trigger_mode = self.settings.trigger_mode
+        state = self.status.state
+
+        if state == State.READY and trigger_mode == "ints":
+            self._set_state(State.ACQUIRE)
+
+            for idx in range(0, self.settings.nimages):
+
+                aquired = Image.create_dummy_image(idx)
+
+                header_json = {
+                    "htype": "dimage-1.0",
+                    "series": "<series id>",
+                    "frame": aquired.index,
+                    "hash": aquired.hash,
+                }
+
+                json2 = {
+                    "htype": "dimage_d-1.0",
+                    "shape": "[x,y,(z)]",
+                    "type": aquired.dtype,
+                    "encoding": aquired.encoding,
+                    "size": len(aquired.data),
+                }
+
+                json3 = {
+                    "htype": "dconfig-1.0",
+                    "start_time": "<start_time>",
+                    "stop_time": "<stop_time>",
+                    "real_time": "<real_time>",
+                }
+
+                LOGGER.debug(header_json)
+                LOGGER.debug(json2)
+                LOGGER.debug(json3)
+
+            return "Aquiring Data from Eiger..."
+        else:
+            return (
+                f"Ignoring trigger, state={self.status.state},"
+                f"trigger_mode={trigger_mode}"
+            )
+
+    async def cancel(self) -> None:
+        """Function to stop the data acquisition.
+
+        Function to stop the data acquisition, but only after the next
+        image is finished.
+        """
+        self._set_state(State.READY)
+
+        header_json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(header_json)
+
+    async def abort(self) -> None:
+        """Function to abort the current task on the Eiger."""
+        self._set_state(State.IDLE)
+
+        header_json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(header_json)
+
+    def update(self, time: SimTime, inputs) -> DeviceUpdate:
```

Comment:
`update` I/O should be typed
```suggestion
    def update(self, time: SimTime, inputs: Inputs) -> DeviceUpdate[Outputs]:
```

Reply:
If i write that I get an error

Reply:
```Argument 2 of "update" is incompatible with supertype "Device"; supertype defines the argument type as "InMap"mypy(error)
This violates the Liskov substitution principlemypy(note)
See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overridesmypy(note)
Return type "DeviceUpdate[Outputs]" of "update" incompatible with return type "DeviceUpdate[OutMap]" in supertype "Device"mypy(error)```

Reply:
and I also get for Inputs:
```Argument 2 of "update" is incompatible with supertype "Device"; supertype defines the argument type as "InMap"mypy(error)
This violates the Liskov substitution principlemypy(note)
See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overridesmypy(note)```

Reply:
Should be resolved with #47